include/benchmark/benchmark.h

```diff
@@ -572,27 +572,27 @@ class State {
   bool finished_;
   size_t total_iterations_;
 
-  std::vector<int> range_;
+  std::vector<int64_t> range_;
 
   size_t bytes_processed_;
   size_t items_processed_;
 
-  int complexity_n_;
+  int64_t complexity_n_;
 
   bool error_occurred_;
 
  public:
   // Container for user-defined counters.
   UserCounters counters;
   // Index of the executing thread. Values from [0, threads).
-  const int thread_index;
+  const int64_t thread_index;
   // Number of threads concurrently executing the benchmark.
-  const int threads;
+  const int64_t threads;
   const size_t max_iterations;
 
   // TODO(EricWF) make me private
-  State(size_t max_iters, const std::vector<int>& ranges, int thread_i,
-        int n_threads, internal::ThreadTimer* timer,
+  State(size_t max_iters, const std::vector<int64_t>& ranges, int64_t thread_i,
+        int64_t n_threads, internal::ThreadTimer* timer,
```

Comment:
i'm not sure we need more than 2^32 threads, do we?

Reply:
I got a bunch of compiler warnings while only changing the `Range` arguments. But yes it is ridiculous to have 64-bit threadcounts.